Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveE.cpp

```diff
@@ -102,51 +131,69 @@ void FiniteDifferenceSolver::MacroscopicEvolveECartesian (
         Box const& tex  = mfi.tilebox(Efield[0]->ixType().toIntVect());
         Box const& tey  = mfi.tilebox(Efield[1]->ixType().toIntVect());
         Box const& tez  = mfi.tilebox(Efield[2]->ixType().toIntVect());
-
-
+        // starting component to interpolate macro properties to Ex, Ey, Ez locations
```

Comment:
I would suggest a different approach in how to include the 1/mu factor on B. Instead of passing in mu to each call of DownwardD[xyz], you can add a loop here that pre-divides B by mu for the whole array and then add a loop afterwards that undoes the divide. This would avoid having to change all of the DownwardD routines. It would also be more efficient since each divide would only be done once whereas with the divides in the DownwardD routines each divide would be done six times.

Reply:
will do :)